chain/dependencies.go

```diff
@@ -113,51 +114,6 @@ type Mempool interface {
 	FinishStreaming(context.Context, []*Transaction) int
 }
 
-type Rules interface {
```

Comment:
As mentioned above, don't move this out of `chain`.

Reply:
We could embed a fee-focused interface into `Rules` but the top-level struct should stay here.

Reply:
I made a fee-focused interface that takes out the functions that `fee_manager` uses from `Rules` and placed it into `fees.Metadata` so we can embed it into `Rules`: 

```
type Metadata interface {
	GetMinUnitPrice() Dimensions
	GetUnitPriceChangeDenominator() Dimensions
	GetWindowTargetUnits() Dimensions
	GetMaxBlockUnits() Dimensions
}
```